src/cursor/cur_hs.c

```diff
@@ -843,9 +859,16 @@ __curhs_update(WT_CURSOR *cursor)
     /* Connect the tombstone to the update. */
     hs_tombstone->next = hs_upd;
 
-    /* Insert the updates and if we fail, search and try again. */
+    /* Make the updates and if we fail, search and try again. */
     while ((ret = __wt_hs_modify(cbt, hs_tombstone)) == WT_RESTART) {
-        WT_WITH_PAGE_INDEX(session, ret = __wt_hs_row_search(cbt, &file_cursor->key, true));
+        WT_WITH_PAGE_INDEX(session, ret = __wt_hs_row_search(cbt, &file_cursor->key, false));
+        WT_ERR(ret);
+        retry = true;
+    }
+
+    /* If we retry, search again to point to the updated value. */
```

Comment:
Why is that only necessary in the retry case?

Reply:
The issue here is that if we retry, the value cursor points to (the new value we just set) is overwritten by the search that finds the old value. We need to research again to point the cursor value to the new value.

Reply:
Makes sense. Thanks.